File: front/src/models/Personnage.ts
import { GradeLeger } from "./Grade"
import { PlaneteOrigine } from "./PlaneteOrigine"
import { SpecialiteLeger } from "./Specialite"

export type Personnage =
{
    id: number,
    nom: string,
    nomDiscord: string,
    matricule: string,
    dateNaissance: string,
    etatService: string | null,
    groupeSanguin: string,
    urlPhotoIdentite: string,
    nbOperation: number,
    listeFormation: string[],
    estFormateur: boolean,
    estFormateurSpecialite: boolean,
    formationFaite: boolean,
    grade: GradeLeger,
    nbOperationGradeBloquer: number,
    nbPointBoutique: number,
    specialite: SpecialiteLeger | null,
    planeteOrigine: PlaneteOrigine | null,
    dateDerniereParticipation: Date | null,
    dateCreation: Date,
    estValider: boolean,
    estNavy: boolean,
    listeMedaille: { id: number, nom: string, urlImage: string }[]
}

export type PersonnageLeger =
{
    id: number,
    nom: string
}

export type PersonnageRequete =
{
    nom: string,
    login: string | null,
    mdp: string | null,
    nomDiscord: string,
    matricule: string,
    groupeSanguin: string,
    etatService: string | null,
    formationFaite: boolean,
    idGrade: number,
    idPlaneteOrigine: number,
    idSpecialite: number | null
}

export type PersonnageModifierRequete =
{
    nom: string,
    nomDiscord: string,
    matricule: string,
    groupeSanguin: string,
    etatService: string | null,
    nbOperation: number,
    listeFormation: string[],
    estFormateur: boolean,
    estFormateurSpecialite: boolean,
    formationFaite: boolean,
    idGrade: number,
    idPlaneteOrigine: number,
    idSpecialite: number | null
}

File: back/back/Models/Personnage.cs
using LiteDB;

namespace back.Models;

public sealed class Personnage
{
    [BsonId]
    public int Id { get; set; }

    [BsonRef]
    public Grade? Grade { get; set; }

    [BsonRef]
    public PlaneteOrigine? PlaneteOrigine { get; set; }

    [BsonRef]
    public Status Status { get; set; } = null!;

    [BsonRef]
    public PersonnageSecondaire? PersonnageSecondaire { get; set; }

    [BsonRef]
    public Specialite? Specialite { get; set; }

     [BsonRef]
     public List<BoutiquePrix> ListeBoutiquePrix { get; set; } = [];

     public string NomDiscord { get; set; } = null!;
    public string Nom { get; set; } = null!;
    public string Matricule { get; set; } = null!;
    public string GroupeSanguin { get; set; } = null!;
    public string? EtatService { get; set; }
    public string? NomFichierPhotoIdentite { get; set; }
    public int NbOperation { get; set; }
    public int NbBootcamp { get; set; }
    public int NbPointBoutique { get; set; }

    public bool EstFormateur { get; set; }
    public bool EstFormateurSpecialite { get; set; }
    public bool FormationFaite { get; set; }

    public DateTime? DateDerniereParticipation { get; set; }
    public DateTime DateCreation { get; set; }
}
